src/index.rs

```diff
@@ -365,33 +369,42 @@ impl ResourceIndex {
             Err(_) => {
                 // updating the index after resource removal is a correct
                 // scenario
-                self.forget_path(canonical_path, old_id)
+                return self.forget_path(canonical_path, old_id);
             }
             Ok(metadata) => {
                 match scan_entry(canonical_path, metadata) {
                     Err(_) => {
                         // a directory or empty file exists by the path
-                        self.forget_path(canonical_path, old_id)
+                        return self.forget_path(canonical_path, old_id);
                     }
                     Ok(new_entry) => {
-                        // valid resource exists by the path
-                        let curr_entry = &self.path2id[canonical_path];
-
-                        if curr_entry.id == new_entry.id {
-                            // in rare cases we are here due to hash collision
+                        // in rare cases we are here due to hash collision
+                        if let Some(curr_entry) =
+                            self.path2id.get(canonical_path)
+                        {
+                            if curr_entry.id == new_entry.id {
+                                if curr_entry.modified == new_entry.modified {
+                                    log::warn!("path {:?} does not modified but not its content", &canonical_path);
+                                    return Ok(update);
+                                }
 
-                            if curr_entry.modified == new_entry.modified {
-                                log::warn!("path {:?} was modified but not its content", &canonical_path);
+                                //in cases Resourceid' didn't modify and 'SystemTime' was modified
+                                // updated
+                                log::warn!("Resourceid didn't modify but SystemTime was modified");
+                                // update.added.insert(
+                                //     canonical_path_buf.clone(),
+                                //     new_entry.id,
+                                // );
+                                // self.insert_entry(
+                                //     canonical_path_buf.clone(),
+                                //     new_entry,
+                                // );
+                                // update.deleted.insert(old_id);
```

Comment:
Commented-out code should be removed before merging.

Reply:
> Commented-out code should be removed before merging.

@kirill I would like to ask you:
Why don't you do anything when the file has changed?

> Commented-out code should be removed before merging.

#kirill I would like to ask you: Why do you want to return "Err" when a file has been modified?
Shouldn't some information be returned when an existing file is modified?